### PR TITLE
update notification modules - add option user_groups

### PR DESCRIPTION
**Pull request by sgruber94**
Adding option user_groups for Notifications

### CODE REVIEW

**Comment by sgruber94** [comment]
Some Sanity Checks will not work, because I can't create a user group automatic ( atm.)

**Comment by rndmh3ro** [comment]
Hi @sgruber94,

thanks for this PR!

Two things: 
* Can you name the variable consistently, please? It should be `user_groups`
* After changing the code in the python module, you do not need to manually alter the tests and examples. Instead follow the steps mentioned here: https://github.com/T-Systems-MMS/ansible-collection-icinga-director#updating-the-tests-and-examples

**Comment by sgruber94** [comment]
I updated already all vars, but didn't push the version
thanks for pointing to the manual. 
I will use that in future :)

-> user_groups will be failing in the tests every time because no user_groups are created automatically

Should i  add Version "added string" or something else?

**Comment by rndmh3ro** [comment]
> I updated already all vars, but didn't push the version

Thanks, however it should be `user_groups`, not `users_groups`.

> Should i add Version "added string" or something else?

Good idea, please add `version_added: '1.16.0'`.

File: plugins/modules/icinga_notification.py [hunk 1]
```diff
@@ -114,6 +114,12 @@
       - When the last notification should be sent.
     type: "int"
     version_added: "1.15.0"
+  user_groups:
+    description:
+      - User Group that should be notified by this notification.
```
Comment (rndmh3ro) [inline plugins/modules/icinga_notification.py:119]:
```suggestion
      - User Groups that should be notified by this notification.
```

File: plugins/modules/icinga_notification_template.py [hunk 2]
```diff
@@ -94,6 +94,12 @@
     type: "list"
     elements: str
     version_added: "1.15.0"
+  user_groups:
+    description:
+      - User Group that should be notified by this notification.
```
Comment (rndmh3ro) [inline plugins/modules/icinga_notification_template.py:99]:
```suggestion
      - User Groups that should be notified by this notification.
```

File: plugins/modules/icinga_notification.py [hunk 3]
```diff
@@ -204,6 +213,7 @@ def main():
         "notification_interval": module.params["notification_interval"],
         "states": module.params["states"],
         "users": module.params["users"],
+        "users_groups": module.params["users_groups"],
```
Comment (rndmh3ro) [inline plugins/modules/icinga_notification.py:216]:
```suggestion
        "user_groups": module.params["user_groups"],
```

File: plugins/modules/icinga_notification.py [hunk 4]
```diff
@@ -175,6 +183,7 @@ def main():
         notification_interval=dict(required=False),
         states=dict(type="list", elements="str", required=False),
         users=dict(type="list", elements="str", required=False),
+        users_groups=dict(type="list", elements="str", required=False),
```
Comment (rndmh3ro) [inline plugins/modules/icinga_notification.py:186]:
```suggestion
        user_groups=dict(type="list", elements="str", required=False),
```

File: plugins/modules/icinga_notification_template.py [hunk 5]
```diff
@@ -169,6 +178,7 @@ def main():
         "period": module.params["time_period"],
         "command": module.params["notification_command"],
         "users": module.params["users"],
+        "user_groups": module.params["users_groups"],
```
Comment (rndmh3ro) [inline plugins/modules/icinga_notification_template.py:181]:
```suggestion
        "user_groups": module.params["user_groups"],
```

**Review by rndmh3ro — CHANGES_REQUESTED** [review]

**Comment by sgruber94** [comment]
@rndmh3ro  please keep in mind, that this  pull request will  break  the automated test ( usergroup OnCall) is not created automatically

**Comment by rndmh3ro** [comment]
> @rndmh3ro please keep in mind, that this pull request will break the automated test ( usergroup OnCall) is not created automatically

Thanks, will do! :)

File: plugins/modules/icinga_notification_template.py [hunk 6]
```diff
@@ -149,6 +157,7 @@ def main():
         time_period=dict(required=False, aliases=["period"]),
         notification_command=dict(required=False, aliases=["command"]),
         users=dict(type="list", elements="str", required=False),
+        users_groups=dict(type="list", elements="str", required=False),
```
Comment (rndmh3ro) [inline plugins/modules/icinga_notification_template.py:160]:
```suggestion
        user_groups=dict(type="list", elements="str", required=False),
```

**Comment by rndmh3ro** [comment]
Thanks, looks good now.
As soon as we have a  module for group-creation, I can merge this.

**Comment by rndmh3ro** [comment]
Hey @sgruber94,

we added support for the user_group module. Can you please merge the master-branch into your branch?

**Comment by sgruber94** [comment]
@rndmh3ro  master branch merged. Also found one typo. And fixed that.

**Review by rndmh3ro — APPROVED** [review]

**Comment by rndmh3ro** [comment]
Thank you!

**Pull request merged**